genestack_client/genestack_metainfo.py

```diff
@@ -147,6 +163,41 @@ def add_person(self, key, name, phone=None, email=None):
         result['email'] = xstr(email)
         self.setdefault(key, []).append(result)
 
+    def add_publication(self, key, title, authors, journal_name,
+                        issue_date, identifiers=None, issue_number=None, pages=None):
+        """
+        Add a publication.
+        All fields will be visible to anyone who has access to this metainfo object.
+
+        :param key:
+        :type key: str
+        :param title: publication title
+        :type title: str
+        :param identifiers: publication identifiers
+        :type identifiers: dict
+        :param authors: publication authors
+        :type authors: str
+        :param journal_name: name of the journal containing this publication
+        :type journal_name: str
+        :param issue_date: journal issue date
+        :type issue_date: str
+        :param issue_number: journal issue number
+        :type issue_number: str
+        :param pages: pages in the journal issue
+        :type pages: str
+        :rtype: None
+        """
+        result = Metainfo._create_dict_with_type('publication')
+        result['identifiers'] = identifiers if identifiers is not None else {}
```

Comment:
Maybe test for `identifiers` emptiness? In case somebody adds empty list, for example.